tests/test_weighted_pandas.py

```diff
@@ -1030,3 +1030,51 @@ def test_read_csv(tmp_path, mcmc_wdf):
     mcmc_wdf.to_csv(filename)
     mcmc_wdf_ = read_csv(filename)
     pandas.testing.assert_frame_equal(mcmc_wdf, mcmc_wdf_)
+
+
+def test_unweighted_dataframe_compression():
+    """Test compression works for unweighted DataFrames with integer."""
+    # Create unweighted data
+    np.random.seed(42)
+    data = WeightedDataFrame(np.random.randn(1000, 3), columns=['A', 'B', 'C'])
+    assert not data.isweighted()
+
+    # Test compression with integer values
+    for n in [100, 200, 500]:
+        compressed = data.compress(n)
+        assert len(compressed) == n
+        assert not compressed.isweighted()
+        assert list(compressed.columns) == list(data.columns)
+
+    # Test compression with values >= original size (should return unchanged)
+    compressed = data.compress(1000)
+    assert len(compressed) == 1000
+
+    compressed = data.compress(1500)
+    assert len(compressed) == 1000
+
+    # Test compression with non-integer values (should return unchanged)
+    compressed = data.compress(True)
+    assert len(compressed) == 1000
+
+    compressed = data.compress('equal')
+    assert len(compressed) == 1000
+
+    compressed = data.compress(False)
+    assert len(compressed) == 1000
```

Comment:
Avoid `for` loops in tests, rather use pytest's parametrize:
```python
@pytest.mark.parametrize('n_requested, n_expected', [(100, 100),
                                                     (500, 500),
                                                     (1000, 1000),
                                                     (1500, 1000),
                                                     (True, 1000),
                                                     (False, 1000),
                                                     ('equal', 1000)])
```

All of these cases can be tested as:
```python
compressed = data.compress(n_requested)
assert len(compressed) == n_expected
assert not compressed.isweighted()
assert list(compressed.columns) == list(data.columns)
```